python/kiss_icp/datasets/rosbag.py

```diff
@@ -55,8 +56,12 @@ def __len__(self):
 
     def __getitem__(self, idx):
         # TODO: implemnt [idx], expose field_names
```

Comment:
The read_messages function in rosbag library returns a Generator Object. It is kind of impossible to implement an indexing based access unless we convert it to a list implicitly.

Exposing the field names is trivial though, we just need to store/return the `msgs.fields` output which is a list of strings containing details of each field in the PointCloud2 message.

Reply:
I see, I actually forgot why I wanted to expose the field_names.

About the sequential thing, I understand. Is not possible, and is not a dream of mine anymore :)